snap/hooks/prepare-device

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+set -eux
+
+exec >> $SNAP_COMMON/prepare-device-hook.log 2>&1
```

Comment:
I'd prefer avoiding any file creation unless this runs for the real use case, and even to return early. Something like
```
if [ -z "$MODEL_APIKEY" ]; then
    exit 0
fi

exec >> $SNAP_COMMON/prepare-device-hook.log 2>&1
...
```